### PR TITLE
MM-43889 Reduce unnecessary calls to viewChannel API

### PR DESCRIPTION
**Pull request by hmhealey**
TL;DR:
- I got rid of the call to viewChannel when unmounting PostList and switching away from a channel
- I got rid of the extra call to viewChannel when switching teams

---

The viewChannel API is called to update the lastViewedAt for a channel as well as keeping track of the channel that the user is active in. This is currently called:
1. When the PostList is mounted, the current channel changes in it, or it's unmounted. This works out to:
    - (mount) When opening the app or switching to a channel
    - (channel change) This doesn't actually occur since we remount the component when this happens
    - (unmount) When switching away from a channel or away from Channels in general
2. When switching teams from the TeamSidebar
3. When focusing the window
4. When blurring the window
    - This is called with an empty channel ID to indicate that you're not active in any channel
6. When receiving a new post while viewing the app
7. When a post is edited in the channel you're viewing
    - I'm honestly not sure why this case is there, but I think it might be something to do with how we sometimes get posts by the lastPostAt when reloading the channel
9. Before the window is unloaded

Most of these use cases have some purpose, but if you're looking closely, you'll notice two problem cases:
1. We make two calls to the API when switching channels (one for the new channel on PostList mount and one for the previous one on PostList unmount)
2. We make those two calls plus a third when switching teams

We used to need to mark the previous channel as read when switching away because we didn't trust that the websocket was reliable. Both used to be done with a single API call, but that caused other issues, so we're not going back to that. Instead, I just removed the call on unmount. Since we mark the channel as read when you view it, when you receive a new message while viewing it, and when you click back onto the window, this should keep it read without the extra call. This is also how mobile works.

For team switching, I just removed the extra call.

The only other case I was looking at removing was the case where you're editing a post. I didn't get to this in the time I had assigned to it, and I think it's a relatively infrequent case that has less performance impact, so I left it as-is.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43889

#### Release Note
```release-note
Reduce number of calls made to viewChannel API during regular usage
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]
Fantastic! Excited for this

cc @lieut-data

**Comment by hmhealey** [comment]
@jgilliam17 This is getting rid of some extra API calls when changing channels/teams for marking a channel as read. One case applies to the previously viewed channel and the other applies to the ones you're switching to. This should hopefully have no effect on the end user since they were duplicates

**Comment by jgilliam17** [comment]
Thanks for running E2Es, [report](https://mattermost-cypress-report.s3.amazonaws.com/194943-1116121-onprem-ent-webapp-pr-10290-mattermost/mm-ee-test1116121/mochawesome.html) looks good, no PR related failures.

**Comment by jgilliam17** [comment]
Tested manually, looks good to merge. 
- Verified switching channels, teams and marking channels as unread/read is working as before.

**Review by jgilliam17 — APPROVED** [review]

**Comment by amyblais** [comment]
@hmhealey Is this for 7.0 or 7.1, or does it matter?

**Comment by lieut-data** [comment]
@amyblais, not sure if you already got an answer, but I think this falls into "it doesn't matter." Just an improvement that can be released whenever :)
